Content.Shared/Nyanotrasen/Shipyard/SharedShipyardSystem.cs

```diff
@@ -0,0 +1,45 @@
+using Content.Shared.Shipyard.Components;
+using Content.Shared.Containers.ItemSlots;
+using JetBrains.Annotations;
+using Robust.Shared.GameStates;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Shipyard
+{
+    [NetSerializable, Serializable]
+    public enum ShipyardConsoleUiKey : byte
+    {
+        //Not currently implemented. Could be used in the future to give other factions a variety of shuttle options.
+        Shipyard,
+        Syndicate
+    }
+
+    [UsedImplicitly]
+    public abstract class SharedShipyardSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<SharedShipyardConsoleComponent, ComponentGetState>(OnGetState);
+            SubscribeLocalEvent<SharedShipyardConsoleComponent, ComponentHandleState>(OnHandleState);
+        }
+
+        private void OnHandleState(EntityUid uid, SharedShipyardConsoleComponent component, ref ComponentHandleState args)
+        {
+            if (args.Current is not ShipyardConsoleComponentState state) return;
+        }
```

Comment:
Just a sanity check to insure proper states are being sent to or from the correct ui, since eventually I will have multiple ui states for syndi/other types of UI catalogues

Reply:
oh this is the handle/get state system for UIs that I only partly understand, but from what I can tell GetState and HandleState are called when generic actions are performed in the UI? I see lots of UI use state.Handled to signify that the call has been handled correctly, and since I don't have any custom logic when handling a state I just put a simple sanity check

Reply:
My limited understanding of ComponentHandleState says that this event is fired on the client side when a component's state (i.e. its variables) changes because the server has told it so (an entity comes into PVS range or needs to be re-synced for example). The server fires ComponentGetState on components when it needs to figure out what state to send to clients. So the flow is something like: server fires GetState, event listeners populate a ComponentState derivative class, then that class is serialized to clients that can know about this state into HandleState.

All of this is only relevant for netsync'd components. If your component has no specific state to send to a client, there's no point in listening to Handle/GetState. This method in particular does nothing; it just returns based on a conditional, but it would've returned anyway. Your event handler for ComponentGetState similarly accomplishes no populating of state; it just creates a derivative class with no information.

I think it's safe to say you could get rid of OnHandleState, OnGetState, and ShipyardConsoleComponentState, because they're not used anywhere. This sort of thing is good if you need the client to know about something and keep that information around without having to constantly poll the server every time you want to know some detail, but based on my initial review, it doesn't seem like you're doing this.